### PR TITLE
Deprecate the STRALGO command and implement the LCS in its place

### LINKED ISSUE #2987: Implement the LCS command and deprecate STRALGO
## Feature Request

Deprecate the STRALGO command and implement the LCS in its place

#### Is your feature request related to a problem? Please describe

With https://github.com/redis/redis/issues/9744 the STRALGO command was deprecated in Redis 7.0
Tests that test the STRALGO command have not been running since the pipeline only uses latest Redis versions

#### Describe the solution you'd like

Implement the [LCS](https://redis.io/docs/latest/commands/lcs/) command

#### Describe alternatives you've considered

For users of the Lettuce driver the only possible solution now is to use the [Custom commands](https://redis.github.io/lettuce/advanced-usage/#custom-commands)

### PR DESCRIPTION
**Pull request by Dltmd202**
Deprecate the STRALGO command and implement the LCS in its place

Is your feature request related to a problem? Please describe
With https://github.com/redis/redis/issues/9744 the STRALGO command was deprecated in Redis 7.0
Tests that test the STRALGO command have not been running since the pipeline only uses latest Redis versions

Describe the solution you'd like
Implement the [LCS](https://redis.io/docs/latest/commands/lcs/) command

Describe alternatives you've considered
For users of the Lettuce driver the only possible solution now is to use the [Custom commands](https://redis.github.io/lettuce/advanced-usage/#custom-commands)

Closes #2987 

Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/lettuce-io/lettuce-core/blob/main/.github/CONTRIBUTING.md).
- [x] You have created a feature request first to discuss your contribution intent. Please reference the feature request ticket number in the pull request.
- [x] You applied code formatting rules using the `mvn formatter:format` target. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.

### CODE REVIEW

**Comment by tishun** [comment]
Hey folks, there seem to be test failures, could you check locally if tests pass?

> make test

**Comment by Dltmd202** [comment]
I'm really sorry.
I thought all I had to do was mvn test, I would test it, edit the code, and upload it quickly.

**Comment by Dltmd202** [comment]
@tishun I modified the code to make the integration test run successfully!

File: src/main/java/io/lettuce/core/LcsArgs.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2011-Present, Redis Ltd. and Contributors
+ * All rights reserved.
+ *
+ * Licensed under the MIT License.
+ *
+ * This file contains contributions from third-party contributors
```
Comment (tishun) [inline src/main/java/io/lettuce/core/LcsArgs.java:7]: FWIW You only need the first 5 lines of the license agreement as this is a new file

File: src/main/java/io/lettuce/core/LcsArgs.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import io.lettuce.core.protocol.CommandArgs;
+
+/**
+ * Argument list builder for the Redis <a href="https://redis.io/commands/lcs">LCS</a> command. Static import the methods from
+ * {@link LcsArgs.Builder} and call the methods: {@code keys(…)} .
+ * <p>
+ * {@link LcsArgs} is a mutable object and instances should be used only once to avoid shared mutable state.
+ *
+ * @author Dltmd202
+ * @since 7.0
```
Comment (tishun) [inline src/main/java/io/lettuce/core/LcsArgs.java:32]: Let's put 6.6 here, It tihnk we can publish this in the next minor release

**Review by tishun — CHANGES_REQUESTED** [review]
Hey, apologies for the late response, there are a few things that are missing I am afraid.

You can get some ideas from the [original PR that introduced STRALGO](https://github.com/redis/lettuce/pull/1282/files), but in short:

- we use a template file to generate the different commands, in this case the `src/main/templates/io/lettuce/core/api/RedisStringCommands.java`, after adding the new command there you can call the tests in the `src/test/java/io/lettuce/apigenerator` to generate the different methods in all related classes. You can comment out the other commands in the "Constants" class to avoid generating anything else but the RedisStringCommands.

File: src/main/java/io/lettuce/core/LcsArgs.java (hunk 2 above)
Comment (Dltmd202) [reply src/main/java/io/lettuce/core/LcsArgs.java:32]: Got it. I'll change all the @since tags in the code I uploaded to 6.6!

**Comment by Dltmd202** [comment]
I realized through the template file (`src/main/templates/io/lettuce/core/api/RedisStringCommands.java`) how it determines the implementation of various connection methods, like reactor and async. 

Additionally, I noticed I had missed the NodeSelection series, so I registered that part as well and re-uploaded the code. 

Thank you so much for your review!

**Comment by tishun** [comment]
> I realized through the template file (`src/main/templates/io/lettuce/core/api/RedisStringCommands.java`) how it determines the implementation of various connection methods, like reactor and async.
> 
> Additionally, I noticed I had missed the NodeSelection series, so I registered that part as well and re-uploaded the code.
> 
> Thank you so much for your review @tishun!

If you run the `apigenerator` test cases they would generate all the required API changes.

Then all you have to do is wire the implementation in.

**Review by tishun — APPROVED** [review]

**Comment by tishun** [comment]
Looks awesome, thanks for the contribution!

**Pull request merged**